miio/gateway.py

```diff
@@ -738,6 +787,21 @@ def get_firmware_version(self) -> Optional[int]:
             )
         return self._fw_ver
 
+    @command()
+    def uninstall_scripts(self):
```

Comment:
this is actual installation of program into gateway memory.... installation and deletion...

Reply:
Could you explain what you can do with loading such a "program" into gateway memory?
Can you only get callbacks when an event happens/property changes or can it do more?

It would be really nice if you could include docstrings """explanation"""" for all new functions that briefly explain what the function can do and how it works.

Reply:
`Could you explain what you can do with loading such a "program" into gateway memory?`
Gateway is able to store limited amount of such "scripts". Scripts are permanent upon restarts. External device call parameters are also encoded with script (i.e. rotate)...
For each devices receiving messages from scripts gateway maintains state... It means gateway is sending ping requests and will not send actual packages in pings are faulty..

`Can you only get callbacks when an event happens/property changes or can it do more?`
Unfortunately we have no callbacks... Script allows to send package to other miio device as a result of action....

Reply:
Subscription mechanism can be implement by scripts if you use target with ip is an ip of python-miio running device... But in xiaomi scripts are not expected to be used as callbacks. It was designed to make gateway implement complete automation listen=>process=>send data to other devices... Such behavior is still possbile. You can create script which turns on wi-fi power plug in cube rotation without processing of callback in hass.